go/ephemeral/lib.go

```diff
@@ -324,24 +324,24 @@ func (e *EKLib) newCacheEntry(generation keybase1.EkGeneration) *teamEKGenCacheE
 }
 
 func (e *EKLib) cacheKey(teamID keybase1.TeamID) string {
-	return string(teamID)
+	return teamID.String()
 }
 
 func (e *EKLib) isEntryValid(val interface{}) (*teamEKGenCacheEntry, bool) {
 	cacheEntry, ok := val.(*teamEKGenCacheEntry)
 	if !ok || cacheEntry == nil {
 		return nil, false
 	}
-	return cacheEntry, (keybase1.TimeFromSeconds(time.Now().Unix()) - cacheEntry.Ctime) < keybase1.TimeFromSeconds(cacheEntryLifetimeSecs)
+	return cacheEntry, time.Now().Sub(cacheEntry.Ctime.Time()) < time.Duration(cacheEntryLifetimeSecs*time.Second)
 }
 
 func (e *EKLib) PurgeTeamEKGenCache(teamID keybase1.TeamID, generation keybase1.EkGeneration) {
 
-	key := e.cacheKey(teamID)
-	val, ok := e.teamEKGenCache.Get(teamID)
+	cacheKey := e.cacheKey(teamID)
+	val, ok := e.teamEKGenCache.Get(cacheKey)
```

Comment:
Was this the bug? The wrong key type here?

Reply:
yeah :(